sw-emulator/lib/periph/src/asym_ecc384.rs

```diff
@@ -672,6 +691,19 @@ impl AsymEcc384 {
         self.iv.as_mut().fill(0);
         self.nonce.as_mut().fill(0);
         self.priv_key_in.as_mut().fill(0);
+        self.dh_shared_key.as_mut().fill(0);
+    }
+
+    /// Generate Diffie-Hellman shared key
+    fn generate_dh_shared_key(&mut self) {
+        let shared_key = Ecc384::compute_shared_secret(
+            &bytes_from_words_le(&self.priv_key_in),
+            &Ecc384PubKey {
+                x: bytes_from_words_le(&self.pub_key_x),
+                y: bytes_from_words_le(&self.pub_key_y),
+            },
+        );
+        self.dh_shared_key = words_from_bytes_le(&shared_key);
```

Comment:
If I am reading the RTL correctly, I think the output will only go to this register if the source was not the key vault, and if the destination was not requested to be the key vault: https://github.com/chipsalliance/caliptra-rtl/blob/25df6d473d61f22a863cc7d2f9016482b7eb5cb4/src/ecc/rtl/ecc_dsa_ctrl.sv#L436

Reply:
So does source and dest being keyvault always go hand in hand?